ddl/ddl_api.go

```diff
@@ -651,35 +659,45 @@ func getDefaultValue(ctx sessionctx.Context, col *table.Column, c *ast.ColumnOpt
 					}
 				}
 				if defaultFsp != fsp {
-					return nil, ErrInvalidDefaultValue.GenWithStackByArgs(col.Name.O)
+					return nil, false, ErrInvalidDefaultValue.GenWithStackByArgs(col.Name.O)
 				}
 			}
 		}
 		vd, err := expression.GetTimeValue(ctx, c.Expr, tp, int8(fsp))
 		value := vd.GetValue()
 		if err != nil {
-			return nil, ErrInvalidDefaultValue.GenWithStackByArgs(col.Name.O)
+			return nil, false, ErrInvalidDefaultValue.GenWithStackByArgs(col.Name.O)
 		}
 
 		// Value is nil means `default null`.
 		if value == nil {
-			return nil, nil
+			return nil, false, nil
 		}
 
 		// If value is types.Time, convert it to string.
 		if vv, ok := value.(types.Time); ok {
-			return vv.String(), nil
+			return vv.String(), false, nil
 		}
 
-		return value, nil
+		return value, false, nil
+	}
+	// handle default next value of sequence. (keep the expr string)
+	str, isSeqExpr, err := handleSequenceDefaultValue(c)
```

Comment:
IMHO it's better to have a function list for all the expressions that need `keep the expr string`.

Reply:
By now only `sequence` does keep the restored expr string.
`now()` func have its own handle logic under the type flag and SQL mode.

Reply:
OK, then please add a comment for `getDefaultValue()`